apps/logging/firehoses.py

```diff
@@ -0,0 +1,82 @@
+import boto3
+import logging
+
+from django.conf import settings
+from rest_framework import status
+from rest_framework.exceptions import APIException
+
+
+"""
+  Firehose class for BFD-Insights connectivity
+"""
+
+
+class BFDInsightsFirehoseException(APIException):
+    # BB2-130 custom exception
+    status_code = status.HTTP_500_INTERNAL_SERVER_ERROR
+
+
+class BFDInsightsFirehoseDeliveryStreamHandler(logging.StreamHandler):
+    """
+    Firehose stream handler class for use in Python logging and /health/firehose health check.
+    """
+    def __init__(self):
+        # By default, logging.StreamHandler uses sys.stderr if stream parameter is not specified
+        logging.StreamHandler.__init__(self)
+
+        self.__firehose = None
+        self.__stream_buffer = []
+
+        try:
+            self.__firehose = boto3.client('firehose')
+        except Exception as err:
+            mesg = "Firehose client initialization failed. err: " + str(err)
+            raise BFDInsightsFirehoseException(mesg)
```

Comment:
Would this result in a 500 response for users?

Reply:
Good call out!   I'll perform some more testing with this to verify.